server.js

```diff
@@ -9,11 +11,102 @@ app.use(cors());
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
 
-app.get("/", function (request, response) {
-  response.send("Hotel booking server.  Ask for /bookings, etc.");
+//Create a new booking
+app.post("/bookings", (req, res) => {
+  const {
+    title,
+    firstName,
+    surname,
+    email,
+    roomId,
+    checkInDate,
+    checkOutDate,
+  } = req.body;
+
+  //check all field been filled up
+  if (
+    !title ||
+    !firstName ||
+    !surname ||
+    !email ||
+    !roomId ||
+    !checkInDate ||
+    !checkOutDate
+  ) {
+    res.send("newBooking has empty field");
+  } //validate email
+  else if (!emailValidator.validate(email)) {
+    res.status(400).json({ msg: "please enter the valid email" });
+  } else if (!moment(checkOutDate).isAfter(checkInDate)) {
+    res.status(400).json({ msg: "checkOutDate is after checkInDate" });
+  } else {
+    const newBooking = req.body;
+    //creating id for the new booking
+
+    newBooking.id = Math.max(...bookings.map((booking) => booking.id), 0) + 1;
+    bookings.push(newBooking);
+    res.json(bookings);
+  }
+});
+//search by date and term
+app.get("/bookings/search", (req, res) => {
+  if (!req.query.term && !req.query.date) return res.send("no date no term");
+  if (req.query.date) {
```

Comment:
you already check this above, maybe you want to check 

`const date = moment(req.query.date, "YYYY-MM-DD")`

that date is a valid date